receiver/googlecloudspannerreceiver/internal/statsreader/projectreader.go

```diff
@@ -33,18 +35,24 @@ func (projectReader *ProjectReader) Shutdown() {
 }
 
 func (projectReader *ProjectReader) Read(ctx context.Context) ([]*metadata.MetricsDataPoint, error) {
-	var result []*metadata.MetricsDataPoint
+	var (
+		result []*metadata.MetricsDataPoint
+		err    error
+	)
 
 	for _, databaseReader := range projectReader.databaseReaders {
-		dataPoints, err := databaseReader.Read(ctx)
-		if err != nil {
-			return nil, err
+		dataPoints, readErr := databaseReader.Read(ctx)
+		err = multierr.Append(err, readErr)
+		if readErr == nil {
+			result = append(result, dataPoints...)
 		}
+	}
 
-		result = append(result, dataPoints...)
+	if len(result) > 0 {
```

Comment:
```suggestion
	if len(result) > 0 && err != nil {
```

I wonder if this would be better in order to bypass handle the normative case in which len(result) > 0 and err == nill